libsolidity/ast/Types.cpp

```diff
@@ -996,7 +999,7 @@ u256 RationalNumberType::literalValue(Literal const*) const
 		auto fixed = fixedPointType();
 		solAssert(fixed, "");
 		int fractionalDigits = fixed->fractionalDigits();
-		shiftedValue = (m_value.numerator() / m_value.denominator()) * pow(bigint(10), fractionalDigits);
+		shiftedValue = m_value.numerator() * pow(bigint(10), fractionalDigits) / m_value.denominator();
```

Comment:
I think we should make a decision on rounding rules here, enforce and document that.

Reply:
It should be as exact as possible, can be converted later if necessary. We shouldn't arbitrarily use information from the rational here.